02_bubble/brvishnu92/main.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	sort := make([]int, len(s))
+	copy(sort, s)
+	swap := true
+	for swap {
+		z := sort[0]
+		swap = false
+		for i, x := range sort {
+			if x < z && i != 0 {
+				swap = true
+				sort[i-1], sort[i] = sort[i], sort[i-1]
+			}
+			z = x
+		}
+	}
+	return sort
+}
+
+func insertion(s []int) []int {
+	sort := make([]int, len(s))
+	copy(sort, s)
+	for i := range sort {
+		for j := i; j > 0; j-- {
+			{
+				if sort[j-1] > sort[j] {
+					sort[j-1], sort[j] = sort[j], sort[j-1]
```

Comment:
this is bubble sort i believe.
insertion sort looks at its core something like this (you find the right spot for an element by shifting the rest):
```go
	for j := 1; j < len(s2); j++ {
		i := j - 1
		inserted := s2[j]
		for i >= 0 && s2[i] > inserted {
			s2[i+1] = s2[i]
			i--
		}
		s2[i+1] = inserted
	}
```

Reply:
Oh yeah, somehow got messed up :-D